File: 1. Week 1 - Barchart/solutions/R/visualization.R
# Visualize Distance per Month using ggplot2

# Load Libraries
if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# 1. Load Data
# Using relative path assuming script is run from the solutions/r directory
data_path <- "../../data/distance_per_month.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # 2. Plot
  # Using geom_col for bar chart (identity)
  p <- ggplot(df, aes(x = Month, y = TotalDistance)) +
    geom_col(fill = "rebeccapurple", width = 0.7) +
    
    # Labels
    labs(
      title = "Total Distance per Month",
      x = "Month",
      y = "Distance (km)"
    ) +
    
    # Theme
    theme_minimal() +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1)
    )
  
  # 3. Save
  ggsave("monthly_distance_chart.png", plot = p, width = 10, height = 6)
  print("Chart saved to monthly_distance_chart.png")
  
} else {
  print(paste("Error: File not found at", data_path))
}

File: 1. Week 1 - Barchart/solutions/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    # Load data
    data_path = '../../data/extract_data.csv' # Assuming standard naming or relative path
    # Actually we should use the Week 1 specific data
    data_path = '../../data/distance_per_month.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    df = pd.read_csv(data_path)
    
    # 1. Plot
    plt.figure(figsize=(10, 6))
    plt.bar(df['Month'], df['TotalDistance'], color='rebeccapurple')
    
    # 2. Labels
    plt.title('Total Distance per Month', fontsize=16)
    plt.xlabel('Month', fontsize=12)
    plt.ylabel('Distance (km)', fontsize=12)
    plt.xticks(rotation=45)
    plt.tight_layout()
    
    # 3. Save
    plt.savefig('monthly_distance_chart.png')
    print("Chart saved to monthly_distance_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()
